### DISCUSSION TITLE
ExcelDNA IntelliSense

### DISCUSSION THREAD

**Discussion by hxhgxy** [Excel-DNA/ExcelDna#626] [Q&A] [answered]
How can I put IntelliSense into my UDF project, and so user just need add one excel addin?
Rather than add two excel addins. Thanks!

**Comment by govert** [comment, marked as answer]
1. Install the **ExcelDna.IntelliSense** package.
2. Add a class that implements `IExcelAddIn` and calls `IntelliSenseServer.Install()` in the `AutoOpen()` implementation.

```
using ExcelDna.Integration;
using ExcelDna.IntelliSense;

namespace MyAddIn
{
    public class AddIn : IExcelAddIn
    {
        public void AutoOpen()
        {
            IntelliSenseServer.Install();
        }

        public void AutoClose()
        {
            IntelliSenseServer.Uninstall();
        }
    }
}
```

**Comment by hxhgxy** [comment]
It's great, thanks!
